contrib/OpenInExplorer.lua

```diff
@@ -32,82 +33,66 @@ Select the photo(s) you wish to find in explorer and press "Go to Folder".
-  return
+local function _(msgid)
+    return gettext.dgettext("OpenInExplorer", msgid)
 end
 
+local act_os = dt.configuration.running_os
+local PS = act_os == "windows" and  "\\"  or  "/"
 
--- FUNCTIONS --
-
-local function open_in_explorer() --Open in Explorer
-  local images = dt.gui.selection()
-  local curr_image = ""
-  if #images == 0 then
-    dt.print('please select an image')
-  elseif #images <= 15 then
-    for _,image in pairs(images) do 
-      curr_image = image.path..PS..image.filename
-      local run_cmd = "explorer.exe /select, "..curr_image
-      dt.print_log("OpenInExplorer run_cmd = "..run_cmd)
-      resp = dsys.external_command(run_cmd)
-    end
-  else
-    dt.print('please select fewer images (max 15)')
-  end
+--Detect OS and modify accordingly--	
+local proper_install = true
+if act_os ~= "macos" and act_os ~= "windows" and act_os ~= "linux" then
+  proper_install = false
+  dt.print_error(_('OpenInExplorer plug-in only supports Windows and Linux at this time'))
+  return
 end
 
-local function open_in_nautilus() --Open in Nautilus
+--The commands to open the OSes' file managers
+local fmng_cmd = {}
+fmng_cmd.linux = [[busctl --user call org.freedesktop.FileManager1 /org/freedesktop/FileManager1 org.freedesktop.FileManager1 ShowItems ass 1 ]]
+fmng_cmd.macos = 'open -R '
+fmng_cmd.windows = 'explorer.exe /select, '
+
+--The working function that opens the file manager with the image file name selected.
+local function open_in_fmanager(os, fmcmd)
   local images = dt.gui.selection()
   local curr_image = ""
   if #images == 0 then
-    dt.print('please select an image')
+    dt.print(_('Please select an image'))
   elseif #images <= 15 then
     for _,image in pairs(images) do 
-      curr_image = image.path..PS..image.filename
-      local run_cmd = [[busctl --user call org.freedesktop.FileManager1 /org/freedesktop/FileManager1 org.freedesktop.FileManager1 ShowItems ass 1 ]] .. df.sanitize_filename("file://"..curr_image) .. [[ ""]]
+      curr_image = df.sanitize_filename(image.path..PS..image.filename)
+      local run_cmd = fmcmd..curr_image
+      if os == 'linux' then run_cmd = run_cmd .. [[ ""]] end
       dt.print_log("OpenInExplorer run_cmd = "..run_cmd)
       resp = dsys.external_command(run_cmd)
     end
   else
-    dt.print('please select fewer images (max 15)')
+    dt.print(_('Please select fewer images (max 15)'))
   end
 end
 
-local function open_in_filemanager() --Open
-  --Inits--
-  if not proper_install then
-    return
-  end
-
-  if (dt.configuration.running_os == "windows") then
-    open_in_explorer()
-  elseif (dt.configuration.running_os == "linux") then
-    open_in_nautilus()
-  end  
-end
 
 -- GUI --
 if proper_install then
   dt.gui.libs.image.register_action(
-    "show in file explorer",
-    function() open_in_filemanager() end,
-    "Opens File Explorer at the selected image's location"
+    _("Show in file explorer"),

```

Comment:
Show should be lower case to fit in with all the other buttons in the selected image module

Reply:
OK.